### PR TITLE
Refactor Dockerfile, use scratch as base image, make static binary

### PR DESCRIPTION
**Pull request by p53**
# Title

## Type

[] Bug fix
[] Feature request
[] Enhancement
[] Docs

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,41 +1,45 @@
+ARG HOMEDIR=/opt/gatekeeper
+
 #
-# Builder image
+# Builder
 #
 
 FROM golang:1.14.4 AS build-env
```
Comment (jangaraj) [inline Dockerfile:7]: I would use current Golang - image `golang:1.15` - so builder uses the latest minor Golang version. Minor versions shouldn't have breaking changes, so it should be safe.

File: Dockerfile [hunk 2]
```diff
@@ -1,41 +1,45 @@
...
-    echo "gatekeeper:x:1000:1000:gatekeeper user:/opt/gatekeeper:/sbin/nologin" >> /etc/passwd && \
-    chown -R gatekeeper:gatekeeper /opt/gatekeeper && \
-    chmod -R g+rw /opt/gatekeeper
-
-COPY templates ./templates
-COPY --from=build-env /gatekeeper ./
-RUN chmod +x gatekeeper
+COPY --from=build-env ${HOMEDIR} ${HOMEDIR}
+COPY --from=build-env /etc/passwd /etc/passwd
+COPY --from=build-env /etc/group /etc/group
```
Comment (jangaraj) [inline Dockerfile:41]: I would copy also CA certificates

File: Dockerfile (hunk 1 above)
Comment (p53) [reply Dockerfile:7]: i would like to update to 15 all things in repo in next release, there is enough changes for now i think in this

File: Dockerfile (hunk 2 above)
Comment (p53) [reply Dockerfile:41]: yup i can copy it there as default, although in serious deployment you would probably need to have mounted some fresh CA's

**Review by ackerleytng — APPROVED** [review]

**Pull request merged**